calico/election.py

```diff
@@ -156,6 +162,36 @@ def _vote(self):
                           "the elected master")
                 self._become_master()
 
+    def _check_master_process(self, master_id):
+        """
+        If the master happens to be on our host, checks if its process is
+        still alive.  If it is not, cleans up the now-stale election key.
+
+        :param master_id: Value loaded from the election key.
+        """
+        # Defensive. In case we ever change the master ID format, only parse
+        # it if it looks like what we expect.
```

Comment:
Good plan - can we have an else arm with just a log in it, in case it isn't? Or (and I just find this cleaner, but that may just be me) : 

    if not match:
         log.badness("AAAARGGGHHHHH")
         return

    <stuff that needs less indentation than before>